File: solutions/csharp/log-levels/1/LogLevels.cs
using System;
using System.Text.RegularExpressions;

static class LogLine
{
    public static string Message(string logLine)
    {
        string replaced = Regex.Replace(logLine, @"\[(ERROR|WARNING|INFO)\]:", "");
        return replaced.Trim();
    }

    public static string LogLevel(string logLine)
    {
        string pattern = @"ERROR|WARNING|INFO";
        Regex regex = new Regex(pattern);
        Match m = regex.Match(logLine);
        return m.ToString().ToLower();
    }

    public static string Reformat(string logLine)
    {
        string message = Message(logLine);
        string level = LogLevel(logLine);
        return $"{message} ({level})";
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>
#include <iostream>

namespace log_line {
std::string message(std::string line) {
    int index = line.find_first_of(':');
    std::string result = line.substr(index+2);
    return result;
}

std::string log_level(std::string line) {
     int start = line.find_first_of('[')+1;
     int end = line.find_first_of(']')-1;
     return line.substr(start, end);
}

std::string reformat(std::string line) {
    std::string level = log_level(line);
    std::string text = message(line);
    std::string result = text + " (" + level + ")"; 
    return result;
}
}  // namespace log_line

